src/main/java/io/fabric8/jenkins/openshiftsync/PipelineJobListener.java

```diff
@@ -263,9 +259,36 @@ private BuildConfigProjectProperty buildConfigProjectForJob(WorkflowJob job) {
 
   private void upsertBuildConfigForJob(WorkflowJob job, BuildConfigProjectProperty buildConfigProjectProperty) {
     boolean create = false;
+
+    logger.info("Finding BuildConfig for namespace: " + buildConfigProjectProperty.getNamespace() + " name: " +
+      buildConfigProjectProperty.getName());
+
     BuildConfig jobBuildConfig = getOpenShiftClient().buildConfigs().
             inNamespace(buildConfigProjectProperty.getNamespace()).withName(buildConfigProjectProperty.getName()).get();
+
+    if (jobBuildConfig == null){
+
+      logger.info("Not able to find BuildConfig for namespace: " + buildConfigProjectProperty.getNamespace() + " name: " +
+        buildConfigProjectProperty.getName());
+
+      logger.info("Finding BuildConfig for namespace: " + buildConfigProjectProperty.getNamespace() +
+        " with label gitRepository: " + buildConfigProjectProperty.getName());
+
+      BuildConfigList jobBuildConfigs = getOpenShiftClient().buildConfigs().
+        inNamespace(buildConfigProjectProperty.getNamespace())
+        .withLabels(Collections.singletonMap(OPENSHIFT_LABELS_BUILD_CONFIG_GIT_REPOSITORY_NAME, buildConfigProjectProperty.getName())).list();
+
+      if (jobBuildConfigs.getItems().size() == 1){
+        jobBuildConfig = jobBuildConfigs.getItems().get(0);
+      }
+
+    }
+
     if (jobBuildConfig == null) {
+
+      logger.info("Not able to find BuildConfig for namespace: " + buildConfigProjectProperty.getNamespace() +
```

Comment:
it would be better to put this log statement in first if block
like 
```
if (jobBuildConfigs.getItems().size() == 1){
....
} else {
 log("not found for gitreposity") // whatever message
}
```

or 

update the log message to

`Creating new build config`